configs/example/gem5_library/checkpoints/simpoints-se-checkpoint.py

```diff
@@ -0,0 +1,70 @@
+from gem5.simulate.exit_event import ExitEvent
+from gem5.simulate.simulator import Simulator
+from gem5.utils.requires import requires
+from gem5.components.boards.simple_board import SimpleBoard
+from gem5.components.memory import DualChannelDDR4_2400
+from gem5.components.processors.simple_processor import SimpleProcessor
+from gem5.components.processors.cpu_types import CPUTypes
+from gem5.isas import ISA
+from gem5.resources.resource import Resource
+from pathlib import Path
+from gem5.components.cachehierarchies.classic.\
+    private_l1_private_l2_cache_hierarchy import(
+    PrivateL1PrivateL2CacheHierarchy
+)
+
+from gem5.components.processors.simpoint import SimPoint
+from gem5.simulate.exit_event_generators import (
+    simpoint_save_checkpoint_generator,
+)
+
+requires(isa_required=ISA.X86)
+
+cache_hierarchy = PrivateL1PrivateL2CacheHierarchy(
+    l1d_size = "32kB",
+    l1i_size="32kB",
+    l2_size="256kB",
+)
+
+memory = DualChannelDDR4_2400(size = "3GB")
```

Comment:
Similarly, we can use `SimpleMemory` to (possibly) improve performance.. Though, TBH, I'm not sure this makes any difference.

Reply:
When I tried to use SimpleMemory, there is an error caused by it not being inherited from the AbstractMemorySystem. It doesn't have the 'get_size' function the stdlib board needs during the its init stage. So I used the SingleChannelDDR3_1600 when taking checkpoints, and changed to DualChannelDDR4_2400 when restoring from a checkpoint. Does it sounds correct?

Reply:
During the testing, I also encountered an error when I had two different sizes setup for the memory when taking checkpoints and restoring from one. Does it mean the memory size has to be the same?

Reply:
Yes and yes 🙂